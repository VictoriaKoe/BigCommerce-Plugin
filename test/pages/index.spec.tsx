import Index from '@pages/index';
import { render, screen, waitFor } from '@test/utils';

jest.mock('@lib/hooks/use-bundles', () => require('@mocks/hooks'));

// Mock the session context
jest.mock('../../context/session', () => ({
    useSession: () => ({
        context: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb250ZXh0IjoidGVzdC1zdG9yZS1oYXNoIn0.test-signature'
    })
}));

// Mock the fetch API
global.fetch = jest.fn();

describe('Homepage', () => {
    beforeEach(() => {
        // Reset the mock before each test
        (fetch as jest.Mock).mockClear();
    });

    test('renders correctly', async () => {
        // Mock the API response
        (fetch as jest.Mock).mockResolvedValueOnce({
            ok: true,
            json: async () => ({
                products: [
                    { id: 1, name: 'Product 1', sku: 'SKU1', variants: [] },
                    { id: 2, name: 'Product 2', sku: 'SKU2', variants: [] }
                ],
                bundles: [
                    { id: 1, name: 'Bundle 1', sku: 'BUNDLE1', isVariant: false, productCount: 2 },
                    { id: 2, name: 'Bundle 2', sku: 'BUNDLE2', isVariant: true, variantName: 'Variant', productCount: 3 }
                ]
            })
        });

        const { container } = render(<Index />);

        // Wait for the component to finish loading
        await waitFor(() => {
            expect(screen.getByText('Product Management')).toBeInTheDocument();
        });

        // Check for the main heading (H1)
        const mainHeading = screen.getByRole('heading', { level: 1 });
        expect(mainHeading).toBeInTheDocument();
        expect(mainHeading).toHaveTextContent('Product Management');

        // Check that the tabs are rendered
        expect(screen.getByText('Products')).toBeInTheDocument();
        expect(screen.getByText('Bundles')).toBeInTheDocument();

        // Check that the table is rendered with products
        expect(screen.getByText('Product 1')).toBeInTheDocument();
        expect(screen.getByText('Product 2')).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });
});
