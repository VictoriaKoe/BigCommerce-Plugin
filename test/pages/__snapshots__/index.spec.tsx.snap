// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders correctly 1`] = `
<div
  class="styled__StyledBox-sc-sj5f1m-0 dnUGcK styled__StyledPanel-sc-1h6ef3q-0 frQUTZ"
>
  <div
    class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 dzErNZ"
  >
    <h2
      class="styled__StyledH2-sc-tqnj75-2 styled__StyledH2-sc-1h6ef3q-1 JBgla iAkOrm"
    >
      Bundle Products
    </h2>
  </div>
  <div
    class="styled__StyledBox-sc-sj5f1m-0 dJATMI styled__StyledFlex-sc-hcvk8l-0 ibZQvs"
  >
    <button
      class="styled__StyledButton-sc-3yq204-0 kuKGmE bd-button"
    >
      <span
        class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
      >
        Refresh Bundles
      </span>
    </button>
  </div>
  <div
    class="styled__StyledBox-sc-sj5f1m-0 kqfsfW"
  >
    <h4
      class="styled__StyledH4-sc-tqnj75-4 rUgdM"
    >
      Bundle: 
      Sample Bundle 1
    </h4>
    <p
      class="styled__StyledText-sc-tqnj75-5 fnzVjl"
    >
      ID: 
      1
    </p>
    <div
      class="styled__StyledBox-sc-sj5f1m-0 dnUGcK styled__StyledPanel-sc-1h6ef3q-0 frQUTZ"
    >
      <div
        class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 dzErNZ"
      >
        <h2
          class="styled__StyledH2-sc-tqnj75-2 styled__StyledH2-sc-1h6ef3q-1 JBgla iAkOrm"
        >
          Bundle Product Quantities
        </h2>
      </div>
      <form>
        <div
          class="styled__StyledBox-sc-sj5f1m-0 BRgwK"
        >
          <p
            class="styled__StyledText-sc-tqnj75-5 fnzVjl"
          >
            Product ID: 
            1
          </p>
          <div>
            <span
              class="styled__StyledInputWrapper-sc-g32raa-0 ivVOAe"
            >
              <div
                class="styled__StyledInputContent-sc-g32raa-3 cklbdD"
              >
                <input
                  class="styled__StyledInput-sc-g32raa-1 hdauTU"
                  id=":r0:"
                  min="1"
                  type="number"
                  value="1"
                />
              </div>
            </span>
          </div>
        </div>
        <div
          class="styled__StyledBox-sc-sj5f1m-0 BRgwK"
        >
          <p
            class="styled__StyledText-sc-tqnj75-5 fnzVjl"
          >
            Product ID: 
            2
          </p>
          <div>
            <span
              class="styled__StyledInputWrapper-sc-g32raa-0 ivVOAe"
            >
              <div
                class="styled__StyledInputContent-sc-g32raa-3 cklbdD"
              >
                <input
                  class="styled__StyledInput-sc-g32raa-1 hdauTU"
                  id=":r1:"
                  min="1"
                  type="number"
                  value="2"
                />
              </div>
            </span>
          </div>
        </div>
        <div
          class="styled__StyledBox-sc-sj5f1m-0 BRgwK"
        >
          <p
            class="styled__StyledText-sc-tqnj75-5 fnzVjl"
          >
            Product ID: 
            3
          </p>
          <div>
            <span
              class="styled__StyledInputWrapper-sc-g32raa-0 ivVOAe"
            >
              <div
                class="styled__StyledInputContent-sc-g32raa-3 cklbdD"
              >
                <input
                  class="styled__StyledInput-sc-g32raa-1 hdauTU"
                  id=":r2:"
                  min="1"
                  type="number"
                  value="1"
                />
              </div>
            </span>
          </div>
        </div>
        <div
          class="styled__StyledBox-sc-sj5f1m-0 gPDrJc"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 bTLtWP"
          >
            <button
              class="styled__StyledButton-sc-3yq204-0 jExzGK bd-button"
              type="submit"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                Update Quantities
              </span>
            </button>
            <button
              class="styled__StyledButton-sc-3yq204-0 kuKGmE bd-button"
              type="button"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                Refresh Page
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    class="styled__StyledBox-sc-sj5f1m-0 kqfsfW"
  >
    <h4
      class="styled__StyledH4-sc-tqnj75-4 rUgdM"
    >
      Bundle: 
      Sample Bundle 2
    </h4>
    <p
      class="styled__StyledText-sc-tqnj75-5 fnzVjl"
    >
      ID: 
      2
    </p>
    <div
      class="styled__StyledBox-sc-sj5f1m-0 dnUGcK styled__StyledPanel-sc-1h6ef3q-0 frQUTZ"
    >
      <div
        class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 dzErNZ"
      >
        <h2
          class="styled__StyledH2-sc-tqnj75-2 styled__StyledH2-sc-1h6ef3q-1 JBgla iAkOrm"
        >
          Bundle Product Quantities
        </h2>
      </div>
      <form>
        <div
          class="styled__StyledBox-sc-sj5f1m-0 BRgwK"
        >
          <p
            class="styled__StyledText-sc-tqnj75-5 fnzVjl"
          >
            Product ID: 
            4
          </p>
          <div>
            <span
              class="styled__StyledInputWrapper-sc-g32raa-0 ivVOAe"
            >
              <div
                class="styled__StyledInputContent-sc-g32raa-3 cklbdD"
              >
                <input
                  class="styled__StyledInput-sc-g32raa-1 hdauTU"
                  id=":r3:"
                  min="1"
                  type="number"
                  value="1"
                />
              </div>
            </span>
          </div>
        </div>
        <div
          class="styled__StyledBox-sc-sj5f1m-0 BRgwK"
        >
          <p
            class="styled__StyledText-sc-tqnj75-5 fnzVjl"
          >
            Product ID: 
            5
          </p>
          <div>
            <span
              class="styled__StyledInputWrapper-sc-g32raa-0 ivVOAe"
            >
              <div
                class="styled__StyledInputContent-sc-g32raa-3 cklbdD"
              >
                <input
                  class="styled__StyledInput-sc-g32raa-1 hdauTU"
                  id=":r4:"
                  min="1"
                  type="number"
                  value="1"
                />
              </div>
            </span>
          </div>
        </div>
        <div
          class="styled__StyledBox-sc-sj5f1m-0 gPDrJc"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 bTLtWP"
          >
            <button
              class="styled__StyledButton-sc-3yq204-0 jExzGK bd-button"
              type="submit"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                Update Quantities
              </span>
            </button>
            <button
              class="styled__StyledButton-sc-3yq204-0 kuKGmE bd-button"
              type="button"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                Refresh Page
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
